server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -925,6 +927,22 @@ static Settings aggregateIndexSettings(
         return indexSettings;
     }
 
+    /**
+     * Updates index settings to set replication strategy by default based on cluster level settings
+     * @param settingsBuilder index settings builder to be updated with relevant settings
+     * @param requestSettings settings passed in during index create request
+     * @param clusterSettings cluster level settings
+     */
+    private static void updateReplicationStrategy(Settings.Builder settingsBuilder, Settings requestSettings, Settings clusterSettings) {
+        if (IndexMetadata.INDEX_HIDDEN_SETTING.get(requestSettings) == true) {
+            settingsBuilder.put(SETTING_REPLICATION_TYPE, ReplicationType.DOCUMENT);
+            return;
+        }
+        if (CLUSTER_REPLICATION_TYPE_SETTING.exists(clusterSettings) && INDEX_REPLICATION_TYPE_SETTING.exists(requestSettings) == false) {
+            settingsBuilder.put(SETTING_REPLICATION_TYPE, CLUSTER_REPLICATION_TYPE_SETTING.get(clusterSettings));
+        }
+    }
```

Comment:
can we set `SETTING_REPLICATION_TYPE` here directly and avoid the condition in IndexSettings?

Reply:
If you are referring to this condition: `INDEX_REPLICATION_TYPE_SETTING.exists(requestSettings) == false` then I think we cannot avoid this condition check here, because we have to make sure that index setting overrride cluster setting and without this condition check that behaviour wouldn't be present